src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,253 @@
+
+        for (Task task : tasks) {
+            try {
+                clearFile();
+
+                String status = "ND";
+                if (task.getStatusIcon().equals("X")) {
+                    status = "D";
+                }
+                fw.write(task.type + " | " + status + " | "
+                        + task.description + System.lineSeparator());
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+        fw.close();
+    }
+
+    public static void clearFile() throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write("");
+        fw.close();
+    }
+
+    public static void readFile() throws FileNotFoundException {
+        Scanner s = new Scanner(taskText);
+        String[] arr;
+
+        while (s.hasNext()) {
+            String text = s.nextLine();
+            arr = text.split("\\|");
+
+            boolean status = arr[1].trim().equals("D");
+
+            Task t;
+            switch (arr[0].trim()) {
+            case "T":
+                t = new Todo(arr[2].trim());
+                t.isDone = status;
+                tasks.add(t);
+
+
+                break;
+            case "D": {
+                String fulltext = arr[2];
+                int index = fulltext.lastIndexOf("(by:");
+                String name = fulltext.substring(0, index).trim();
+                String by = fulltext.substring(index + 4, fulltext.length() - 1).trim();
+
+                t = new Deadline(name, by);
+                t.isDone = status;
+                tasks.add(t);
+
+                break;
+            }
+            case "E": {
+                String fulltext = arr[2];
+                int index = fulltext.lastIndexOf("(at:");
+                String name = fulltext.substring(0, index).trim();
+                String at = fulltext.substring(index + 4, fulltext.length() - 1).trim();
+
+                t = new Event(name, at);
+                t.isDone = status;
+                tasks.add(t);
+                break;
+            }
+            }
+
+        }
+        s.close();
+    }
+
+    /*MAIN*/
+
+    public static void main(String[] args) throws DukeException, IOException {
+
+        printStartMessage();
+
+        try {
+            readFile();
+        } catch (FileNotFoundException e){
+            System.out.println("File not found");
+        }
+
+        String input, error;
+        Scanner in = new Scanner(System.in);
+
+        Task t;
+        boolean isBye = false;
+//        File taskList = new File(filePath);
+
+        while (!isBye) {
+            input = in.nextLine();
+
+            if (input.equalsIgnoreCase(("bye"))) { //end program
+                isBye = true;
+                printByeMessage();
+
+            } else if (input.equalsIgnoreCase("list")) {
+                printList(); //print duke.task list
+
+            } else if (input.contains("done")) {
```

Comment:
This might be a bit buggy: what if I want to add a task "Explore aban**done**d house" or "Make a well-**done** steak"? Similar for all other parsing methods which use *contains* to check